rasterio/_base.pyx

```diff
@@ -596,14 +613,9 @@ cdef class DatasetReader(object):
         cdef void *hobj
         cdef const char *domain_c
         cdef char **papszStrList
-        if self._hds == NULL:
-            raise ValueError("can't read closed raster file")
```

Comment:
@perrygeo @brendan-ward I'm yanking all these NULL pointer checks as I can't imagine ever getting into a state where the reference is NULL and our program hasn't already :hankey:d its pants.